src/orion/benchmark/assessment/averagerank.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.benchmark.assessment` -- Average Rank Assessment
+================================================================
+
+.. module:: assessment
+   :platform: Unix
+   :synopsis: Benchmark algorithms with average rank.
+
+"""
+
+from collections import defaultdict
+
+import numpy as np
+import pandas as pd
+import plotly.express as px
+
+from orion.benchmark.base import BaseAssess
+
+
+class AverageRank(BaseAssess):
+    """
+    For each algorithm, run fixed number of Experiment, average the rank of trials for
+    the same algorithm at the same trial sequence order.
+    For the performance of trials in an Experiment, instead using the actual trial objective value,
+    here we use the best objective value in the same Experiment until the particular trial.
+    """
+
+    def __init__(self, task_num=1):
+        super(AverageRank, self).__init__(task_num=task_num)
+
+    def plot_figures(self, task, experiments):
+        """
+        Generate a `plotly.graph_objects.Figure`
+
+        task: str
+            Name of the task
+        experiments: list
+            A list of (task_index, experiment), where task_index is the index of task to run for
+            this assessment, and experiment is an instance of `orion.core.worker.experiment`.
+        """
+        task_algorithm_exp = defaultdict(list)
+
+        for task_index, exp in experiments:
+            algorithm_name = list(exp.configuration["algorithms"].keys())[0]
+
+            trials = list(
+                filter(lambda trial: trial.status == "completed", exp.fetch_trials())
+            )
+            exp_trails = self._build_exp_trails(trials)
```

Comment:
Looks like you are building list of best objectives found in submit time order. You could reuse the `regret` function here instead: https://github.com/Epistimio/orion/blob/develop/src/orion/analysis/regret.py#L14. I think working on a panda frame will make it easier for the work needed in `_display_plot`.

Reply:
Also, I think it may be better if we extract the plotting functions and move them to the plotting backing. It would be great if we could generalize the regret function [here](https://github.com/Epistimio/orion/blob/develop/src/orion/plotting/backend_plotly.py#L139) to plot multiple experiments together.

Reply:
Let me see if pandas can make it easy :)

At this moment, one reason for the existence of the assessment is to draw different charts for different assess purpose, if we do this some plot can be shared, we can put it into the some utils code.